### PR TITLE
Add forceUpdate to updateItem

### PR DESCRIPTION
**Pull request by jmjoary**
Allow to force the update even if there is no change in the object. This boolean is set to false by default.

### CODE REVIEW

File: lib/src/directus_api_manager_base.dart [hunk 1]
```diff
@@ -389,17 +392,21 @@ class DirectusApiManager implements IDirectusApiManager {
 
   @override
   Future<Type> updateItem<Type extends DirectusData>(
-      {required Type objectToUpdate, String? fields}) {
+      {required Type objectToUpdate,
+      String? fields,
+      bool forceSaving = false}) {
```
Comment (maxbritto) [inline lib/src/directus_api_manager_base.dart:397]: I think I'd prefer the name `force` which is more commonly used. 
Also could you add some doc comments to the function and its parameters ? since you deprecated the `updateDurectusUser` rerouting it here, it would help everyone how this function should be used.

File: test/directus_api_test.dart [hunk 2]
```diff
@@ -142,7 +142,7 @@ void main() {
               operator: FilterOperator.between,
               value: [r"$NOW", r"$NOW(+2 weeks)"]));
       expect(request.url.toString(),
-          r'http://api.com/items/article?fields=*&filter=%7B+%22date%22%3A+%7B+%22_between%22%3A+%5B%22%24NOW%22%2C+%22%24NOW%28%2B2+weeks%29%22%5D+%7D%7D');
+          r'http://api.com/items/article?fields=*&filter=%7B+%22date%22%3A+%7B+%22_between%22%3A+%5B%22%24NOW%22%2C%22%24NOW%28%2B2+weeks%29%22%5D+%7D%7D');
```
Comment (maxbritto) [inline test/directus_api_test.dart:145]: What is this change, and how is this related to the current PR ?

**Review by maxbritto — COMMENTED** [review]
Thanks @jmjoary for the PR, that's an interesting addition 👍
I just have a few feedback before I can merge it.

File: lib/src/directus_api_manager_base.dart (hunk 1 above)
Comment (jmjoary) [reply lib/src/directus_api_manager_base.dart]: This is done in e7fc92b51b105d5831d5f0e99335feaa407e58ea

File: test/directus_api_test.dart (hunk 2 above)
Comment (jmjoary) [reply test/directus_api_test.dart:145]: The unit test wasn't passing since bdc0b7e14284e2686da28ed720ed539c14e71143. It has no relation with the main goal of this PR, but I prefer when the tests pass. Do you wish an other PR for that ?
